packages/gatsby-remark-autolink-headers/src/gatsby-ssr.js

```diff
@@ -52,10 +62,14 @@ exports.onRenderBody = ({ setHeadComponents }, pluginOptions) => {
     })
   `
 
+  const style = showIcon
+    ?  <style key={`gatsby-remark-autolink-headers-style`} type="text/css">
+        {styles}
+      </style>
+    : undefined
+
   return setHeadComponents([
-    <style key={`gatsby-remark-autolink-headers-style`} type="text/css">
-      {styles}
-    </style>,
+    style,
```

Comment:
Have you validated how `undefined` works here? I think it'll be ignored, but one easy way to validate is with `gatsby-dev-cli`. Check out the [contributor's guide](https://www.gatsbyjs.org/docs/how-to-contribute/) for more info!

Reply:
@jonsuh you're the MVP :)

Thanks! I'll get this merged and pushed in a bit!